File: TimelineView/src/main/java/com/yogeshpaliyal/timelineview/TimelineListener.java
package com.yogeshpaliyal.timelineview;

import com.yogeshpaliyal.timelineview.interfaces.TimelineAvailability;

import java.util.Calendar;

public interface TimelineListener {
    void onDateChange();

    void parentScrollTo(int y);

    void onStartChange(Calendar calendar);

    void onEndChange(Calendar calendar);

    void onCancelClicked(TimelineAvailability bookedSlots);

    void onPublishStatusChange(boolean isEnable);
}

File: TimelineView/src/main/java/com/yogeshpaliyal/timelineview/TimelineListener.kt
package com.yogeshpaliyal.timelineview

import com.yogeshpaliyal.timelineview.interfaces.TimelineAvailability

interface TimelineListener {
    fun onDateChange()
    fun parentScrollTo(y: Int)
    fun onStartChange(minute: Int)
    fun onEndChange(minute: Int)
    fun onCancelClicked(bookedSlots: TimelineAvailability?)
    fun onPublishStatusChange(isEnable: Boolean)
}